src/components/Resource.js

```diff
@@ -128,6 +136,23 @@ export default class Resource extends React.Component {
     }
   }
 
+  isUpdatedResource(id) {
+    this.state.resources.forEach((i) => {
+      if (i.id === id) {
+        i.upvotes = i.upvotes - 1;
+        i.veteran_has_upvoted = false;
+      }
+    })
+  }
+
+  indexToUpdate(resourceId) {
```

Comment:
perhaps when the resources are all retrieved you can do a preprocess and store a map between the resource ID to the index that it belongs in your list so you don't have to do this every time.

Reply:
if we have a million resources it could take a long time